aws/validators.go

```diff
@@ -2427,6 +2427,11 @@ var validateCloudWatchEventArchiveName = validation.All(
 	validation.StringMatch(regexp.MustCompile(`^[\.\-_A-Za-z0-9]+`), ""),
 )
 
+var validateMQBrokerName = validation.All(
+	validation.StringLenBetween(1, 50),
+	validation.StringMatch(regexp.MustCompile(`^[0-9A-Za-z._~-]+$`), ""),
```

Comment:
When I've tried in the AWS Console, it doesn't accept `~` or `.`

Reply:
Good spot, thank you! Here is a relevant snippet from [Amazon MQ AWS API docs](https://docs.aws.amazon.com/amazon-mq/latest/api-reference/brokers.html#brokers-model-createbrokerinput) for reference:
> Required. The broker's name. This value must be unique in your AWS account, 1-50 characters long, must contain only letters, numbers, dashes, and underscores, and must not contain white spaces, brackets, wildcard characters, or special characters.

I've fixed the regex and added info to the docs in 228d529.